gazebo_ros/src/gazebo_ros_factory.cpp

```diff
@@ -108,7 +108,9 @@ GazeboRosFactory::~GazeboRosFactory()
 
 void GazeboRosFactory::Load(int argc, char ** argv)
 {
-  rclcpp::init(argc, argv);
+  if (!rclcpp::is_initialized()) {
+    rclcpp::init(argc, argv);
+  }
```

Comment:
Good call! It may also be worth it adding a check to `gazebo_ros_init`, and in both cases, printing a warning if it's been initialized, so the user knows their arguments won't be loaded.

Reply:
Ok, let's go without the warning, thanks!